src/Analysis/Ast/Test/TypingTests.cs

```diff
@@ -342,6 +342,39 @@ class A:
                 .Which.Should().HaveType("Tuple[None, None, None]");
         }
 
+        [TestMethod, Priority(0)]
+        public async Task UnionOfTuples() {
+            const string code = @"
+from typing import Union, Tuple, Type
+
+class TracebackType: ...
+
+_ExcInfo = Tuple[Type[BaseException], BaseException, TracebackType]
+_OptExcInfo = Union[_ExcInfo, Tuple[None, None, None]]
+
+x: _ExcInfo
+y: _OptExcInfo
+
+a, b, c = y
+";
+            var analysis = await GetAnalysisAsync(code, PythonVersions.LatestAvailable3X);
+
+            analysis.Should().HaveVariable("x")
+                .Which.Should().HaveType("Tuple[Type[BaseException], BaseException, TracebackType]");
+
+            analysis.Should().HaveVariable("y")
+                .Which.Should().HaveType("Union[Tuple[Type[BaseException], BaseException, TracebackType], Tuple[None, None, None]]");
+
+            analysis.Should().HaveVariable("a")
+                .Which.Should().HaveType("Union[Type[BaseException], None]");
+
+            analysis.Should().HaveVariable("b")
+                .Which.Should().HaveType("Union[BaseException, None]");
+
+            analysis.Should().HaveVariable("c")
+                .Which.Should().HaveType("Union[TracebackType, None]");
```

Comment:
It'd be nice some day to convert `Union[X, None]` to `Optional[X]`, since that's the more readable choice, but this is correct for now.